### ISSUE TITLE
ipython REPL read only [ob-ipython-upstream]

### ISSUE THREAD

**Issue by roblem** [jkitchin/scimax#166]
Accessing the ipython REPL using either `M - DownArrow` or `C-c C-v C-z` opens the ipython repl but it is read only (`funcall-interactively: Text is read-only`) so no code evaluation is possible.  Seems like this was working a few days ago.  Also, I am testing against a local kernel.

**Comment by jkitchin** [comment]
I also thought it was working a few days ago, and now I also see it isn't. I don't know what I could have changed to make that happen though. I guess a bisect on commits might find it.

**Comment by roblem** [comment]
Yeah, I looked through the recent commits and didn't see anything obvious.  The obvious ones are ob-ipython upstream and elpy,  I didn't see any bug reports there for this issue.

**Comment by jkitchin** [comment]
I think I found the issue in commit ec4ad50. It is not obvious at all, but it seems like using button-lock is the culprit. That was a neat idea, but maybe it will need another strategy.

**Comment by roblem** [comment]
Yes.  That fixed it.

**Comment by roblem** [comment]
I didn't actually try and use it yesterday.  The buffer is no longer read only, but when I run code like this:
```python
#+BEGIN_SRC ipython :session 
a = 1
print(a)
#+END_SRC

#+RESULTS:
:RESULTS:
# Out[1]:
# output
: 1
:END:
```
and I switch to the REPL, and run something like this
```python
print(a)
```
I get this:
```
NameErrorTraceback (most recent call last)
<ipython-input-3-bca0e2660b9f> in <module>()
----> 1 print(a)

NameError: name 'a' is not defined
```

**Comment by jkitchin** [comment]
That error is caused by the :session in the header.  you don't need that, because it is defined as a default header arg anyway.

The default is to create a unique kernel for each buffer. It doesnt seem to work right when you define a :session in the header 

If you run this to disable the unique kernel:

#+BEGIN_SRC emacs-lisp
(setq ob-ipython-buffer-unique-kernel nil)
#+END_SRC

your block should work fine.
